### ISSUE TITLE
Preferences window shares the same Window Controller

### ISSUE THREAD

**Issue by cjdowner** [balancemymoney/balance-open#207]
Slightly odd one here (0.95 beta). The Preferences window doesn't behave as expected (compared to other apps on macOS).

Steps to reproduce:
Launch Balance, and then open the Preferences window either via the menu, or shortcut.
Click the Balance icon in the Menu Bar to dismiss the app window
Click it again to reveal the window

As you can see, these two are tied, and this has some consequences on behaviour. For example, if you do the Shift-Command-4-Space shortcut to take a screenshot of either window, both will be included.

I also suspect this is one of the reasons why the Preferences window doesn't disappear when you press Add Account in the accounts view.

There's further behaviour that makes this weird:
- Have both the app window, and the pref window open,
- Minimize the pref window to the doc (yellow traffic light)
- (The app window hides)
- Click the Balance icon in the menubar to show the app window
- (The preferences window appears, but there's still a preview of it in the Dock)
- Click the Prefs window preview in the dock, the window will animate out

Note - not sure if "window controller" is the right term... is it the delegate or outlet? Hope the report is detailed enough that you understand what I mean anyway.

**Comment by cjdowner** [comment]
As a related issue, the window's appearance is quite aggressive. If I open the Balance window, and then switch to another app - that app will be active - however it will sit behind the the Balance window (and open Pref window, if it is indeed open).

The Dropbox app is a good example of a menubar app whose window correctly hides when you have switched to another app.

**Comment by einsteinx2** [comment]
They're definitely different window controllers, but we'll look into this behavior. 

Regarding your additional comment, if you are seeing that behavior without preferences open then you may have the keepPopoverPinned debug user default enabled and forgot. 

Run the following in Terminal to disable it:
`defaults write software.balanced.balance-open keepPopoverPinned -bool false`

Note that we pin the app window when preferences is open so that the app doesn't disappear when you click within the prefs window. This (and essentially all of the issues described in this issue) are related to the popover library we use and have been issues since the original Balance app. When we have time we either need to switch popover libraries or do some smoothing of the behavior here. 

Moving to 1.1

**Comment by cjdowner** [comment]
Great - that command has made things better 👍. I don't think it was something I had played with previously though.

>Note that we pin the app window when preferences is open so that the app doesn't disappear when you click within the prefs window. 

Ah yes. That makes a load of sense.

**Comment by cjdowner** [comment]
Related issue: In the case that Balance throws up an alert, for example – cannot write a shortcut, the UIAlertView actually appears under the preference window.

<img width="526" alt="screen shot 2017-12-13 at 17 27 35" src="https://user-images.githubusercontent.com/2099705/33953172-07e8b104-e02c-11e7-93a0-8ab52d01838f.png">

**Comment by einsteinx2** [comment]
Good catch. This is pretty low priority though so keeping in backlog for now.
